### PR TITLE
Allow unordered metadata

### PR DESCRIPTION
**Pull request by kalaspuffar**
Hi @josteinaj 

Trying to solve https://github.com/nlbdev/nordic-epub3-dtbook-migrator/issues/536

This is a small change moving the uniqueness check of the metadata fields to Schematron.

I created a new rule called nordic_1. Maybe this number is already used, then we will change it, but validating unique fields in any order is not allowed in RelaxNG.

I did a quick check if I could add the Schematron rules as an extension of the RelaxNG rules. Seemed not to work so to be careful I created a new rule.

### CODE REVIEW

File: src/main/resources/xml/schema/2020-1/nordic2020-1.sch [hunk 1]
```diff
@@ -10,6 +10,17 @@
...
     <ns prefix="prism" uri="http://www.prismstandard.org/specifications/3.0/PRISM_CV_Spec_3.0.htm#"/>
 
+    <pattern id="epub_nordic_1">
+        <title>Rule 1</title>
+        <p>Only allow one charset, identifier, title and viewport</p>
+        <rule context="html:head">
+            <assert test="count(html:meta[@charset='UTF-8'])=1">[nordic01] charset seems not to be in head exactly once</assert>
+            <assert test="count(html:meta[@name='dc:identifier'])=1">[nordic01] dc:identifier seems not to be in head exactly once</assert>
+            <assert test="count(html:title)=1">[nordic01] title seems not to be in head exactly once</assert>
+            <assert test="count(html:meta[@name='viewport'])=1">[nordic01] viewport seems not to be in head exactly once</assert>
```
Comment (josteinaj) [inline src/main/resources/xml/schema/2020-1/nordic2020-1.sch:20]: @kalaspuffar could you rephrase these to be "must be" instead of "seems not to be"? Just for consistency with other assertions.

Also, I don't think you need to require at least one title element and charset meta element, since I don't think they are required by the nordic standard.
- Missing title elements should ideally be picked up by epubcheck and/or Ace.
- And for missing charset; I guess UTF-8 would be implicit? I have not checked.

**Review by josteinaj — CHANGES_REQUESTED** [review]

File: src/main/resources/xml/schema/2020-1/nordic2020-1.sch (hunk 1 above)
Comment (kalaspuffar) [reply src/main/resources/xml/schema/2020-1/nordic2020-1.sch]:
```
2.5.2 Title
The <title> element of every xhtml content document must match the dc:title metadata of the package file.

2.5.3 Metadata
The content documents are required to contain two <meta> elements.

<meta name="dc:identifier" content="_[production UID]_"/>
<meta name="viewport" content="width=device-width"/>
The production UID must match the dc:identifier in package.opf.
```

You are correct that there is no mention of either title or charset. But they were validated in the original rules so I didn't want to remove them when I changed to validate using Schematron. But if we feel confidable with removing them, then we could do it.

Not supplying a title or charset is something I would strongly not recommend, as they will ensure some structure. And I believe that EPUBCheck will even fail if you supply any other value for charset than UTF-8.

Comment (kalaspuffar) [reply src/main/resources/xml/schema/2020-1/nordic2020-1.sch]:
```
[ace] Document does not have a non-empty &lt;title&gt; element<br/><br/>&lt;html xmlns=&quot;http://www.w3.org/1999/xhtml&quot; xmlns:epub=&quot;http://www.idpf.org/2007/ops&quot; xml:lang=&quot;en&quot; lang=&quot;en&quot;&gt; (media_overlay_test-001-foreword.xhtml)
[epubcheck] Line: -1 Col: -1 Error while parsing file: assertion failed: WARNING: The &quot;head&quot; element should have a &quot;title&quot; child element. (EPUB/media_overlay_test-001-foreword.xhtml)
[nordic01] charset seems not to be in head exactly once (EPUB/media_overlay_test-001-foreword.xhtml)
[nordic01] title seems not to be in head exactly once (EPUB/media_overlay_test-001-foreword.xhtml)
```

I've rerun it, and the title is validated with the EPUBCheck and the Ace rules, but they do not require the charset.
